gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergRegisterStepTest.java

```diff
@@ -75,6 +75,48 @@ protected IcebergCatalog createDestinationCatalog() throws IOException {
     }
   }
 
+  @Test
+  public void testRegisterIcebergTableWithRetryer() throws IOException {
+    TableMetadata justPriorDestTableMetadata = mockTableMetadata("foo", "bar");
+    TableMetadata readTimeSrcTableMetadata = Mockito.mock(TableMetadata.class);
+    IcebergTable mockTable = mockIcebergTable("foo", "bar");
+    Mockito.doThrow(new RuntimeException()).doThrow(new RuntimeException()).doNothing().when(mockTable).registerIcebergTable(any(), any());
+    IcebergRegisterStep regStep =
+        new IcebergRegisterStep(srcTableId, destTableId, readTimeSrcTableMetadata, justPriorDestTableMetadata, new Properties()) {
+          @Override
+          protected IcebergCatalog createDestinationCatalog() throws IOException {
+            return mockSingleTableIcebergCatalog(mockTable);
+          }
+        };
+    try {
+      regStep.execute();
+      Mockito.verify(mockTable, Mockito.times(3)).registerIcebergTable(any(), any());
+    } catch (RuntimeException re) {
+      Assert.fail("Got Unexpected Runtime Exception", re);
+    }
+  }
+
+  @Test
+  public void testRegisterIcebergTableWithRetryerThrowsRuntimeException() throws IOException {
+    TableMetadata justPriorDestTableMetadata = mockTableMetadata("foo", "bar");
+    TableMetadata readTimeSrcTableMetadata = Mockito.mock(TableMetadata.class);
+    IcebergTable mockTable = mockIcebergTable("foo", "bar");
+    Mockito.doThrow(new RuntimeException()).when(mockTable).registerIcebergTable(any(), any());
+    IcebergRegisterStep regStep =
+        new IcebergRegisterStep(srcTableId, destTableId, readTimeSrcTableMetadata, justPriorDestTableMetadata, new Properties()) {
+          @Override
+          protected IcebergCatalog createDestinationCatalog() throws IOException {
+            return mockSingleTableIcebergCatalog(mockTable);
+          }
+        };
+    try {
+      regStep.execute();
+      Assert.fail("Expected Runtime Exception");
+    } catch (RuntimeException re) {
+      Assert.assertTrue(re.getMessage().startsWith("Failed to register iceberg table (retried"), re.getMessage());
```

Comment:
let's combine this test w/ that of passing overriding retryer config.  e.g. specify that it should retry N times and then look for that number in the message text